stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
     """
 
-    rank = Attribute(
-        int
-    )  #: Rank in segment (either overall leaderboard, or pr rank)
-    type = Attribute(
-        str
-    )  #: The type of achievement -- e.g. 'year_pr' or 'overall'
-    type_id = Attribute(
-        int
-    )  #: Numeric ID for type of achievement?  (6 = year_pr, 2 = overall ??? other?)
+    type_id: Optional[int] = None
+    """
+    Numeric ID for type of achievement?  (6 = year_pr, 2 = overall ??? other?)
+    """
+
+    effort_count: Optional[int] = None
+
 
 
-class BaseEffort(LoadableEntity):
+class BaseEffort(DetailedSegmentEffort, BackwardCompatibilityMixin, DeprecatedSerializableMixin):
     """
     Base class for a best effort or segment effort.
     """
 
-    name = Attribute(str, (SUMMARY, DETAILED))  #: The name of the segment
-    segment = EntityAttribute(
-        Segment, (SUMMARY, DETAILED)
-    )  #: The associated :class:`stravalib.model.Segment` for this effort
-    activity = EntityAttribute(
-        "Activity", (SUMMARY, DETAILED)
-    )  #: The associated :class:`stravalib.model.Activity`
-    athlete = EntityAttribute(
-        Athlete, (SUMMARY, DETAILED)
-    )  #: The associated :class:`stravalib.model.Athlete`
-    kom_rank = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: 1-10 segment KOM ranking for athlete at time of upload
-    pr_rank = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: 1-3 personal record ranking for athlete at time of upload
-    moving_time = TimeIntervalAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.timedelta`
-    elapsed_time = TimeIntervalAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.timedelta`
-    start_date = TimestampAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.datetime` when effort was started in GMT
-    start_date_local = TimestampAttribute(
-        (SUMMARY, DETAILED), tzinfo=None
-    )  #: :class:`datetime.datetime` when effort was started in activity timezone for this effort
-    distance = Attribute(
-        int, (SUMMARY, DETAILED), units=uh.meters
-    )  #: The distance for this effort.
-    average_watts = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: Average power during effort
-    device_watts = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  #: True if the watts are from a power meter, false if estimated
-    average_heartrate = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: Average HR during effort
-    max_heartrate = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: Max HR during effort
-    average_cadence = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: Average cadence during effort
-    start_index = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: The activity stream index of the start of this effort
-    end_index = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: The activity stream index of the end of this effort
+    _field_conversions = {
```

Comment:
another question (bare with me!). 
So here i see the backwardCompatibilityMixin is used and the deprecated mixin is used too. Does that mean that this endpoint is allowing for backwards compatibility. but if so how does backwards compat  work if all of the attributes that were there are removed? 

and why would it be deprecated if the api returns data? i may just not understand. 

is it deprecated because we are deprecating the WAY someone accesses the data via stravalib? but the api still returns this information.

Reply:
> but if so how does backwards compat work if all of the attributes that were there are removed?

See the [previous conversations](https://github.com/stravalib/stravalib/pull/322#discussion_r1083843237). The attributes are inherited from the generated model.

The `DeprecatedSerializableMixin` allows us to deprecate (but still support for the time being) the legacy (de-)serialization methods `deserialize()`, `from_dict()`, and `to_dict()`. It is not deprecating the entity itself or any endpoint returning these entities, but just these three (de-)serialization methods.

Reply:
got it. and yes again i see i'm asking similar questions :)